### PR TITLE
feat: Multiple agent support in CLI

### PR DESCRIPTION
**Pull request by meritmalling**
This PR updates the CLI to support deploying multiple agents per environment.

New commands:
- list: lists all agents in the env
- delete: deletes a given agent

Updated commands:
- deploy: adds -s (slug) option (can be set by env variable as well)
- develop: confirms develop when there are deployed agents 

`npx flatfile list`

Screenshot 2024-02-21 at 6 02 11 PM

`npx flatfile delete`

Screenshot 2024-02-21 at 6 03 57 PM

`npx flatfile deploy -s <slug>`

Screenshot 2024-02-21 at 6 03 22 PM

`npx flatfile deploy` (no slug passed, but multiple listeners already present)

Screenshot 2024-02-21 at 6 03 01 PM

`npx flatfile develop` (when an environment has agents it requests confirmation to proceed)

Screenshot 2024-02-21 at 6 11 11 PM

- Also adds an extractor listener example

### CODE REVIEW

File: packages/cli/src/shared/constants.ts [hunk 1]
```diff
@@ -24,13 +24,21 @@ export const deployTopics = [
   'document:updated',
   'document:deleted',
 
+<<<<<<< HEAD
```
Comment (bangarang) [inline packages/cli/src/shared/constants.ts:27]: Looks like some Merge Conflicts in here... 😢

File: packages/cli/src/x/actions/deploy.action.ts [hunk 2]
```diff
@@ -9,17 +9,24 @@ import { apiKeyClient } from './auth.action'
...
 // @ts-expect-error
 import ncc from '@vercel/ncc'
-import { deployTopics } from '../../shared/constants'
+import { deployTopics, tableConfig } from '../../shared/constants'
 import { getAuth } from '../../shared/get-auth'
 import { getEntryFile } from '../../shared/get-entry-file'
 import { messages } from '../../shared/messages'
+import { agentTable } from '../helpers/agent.table'
+import prompts from 'prompts'
+import sanctuary from 'sanctuary'
```
Comment (bangarang) [inline packages/cli/src/x/actions/deploy.action.ts:22]: `sanctuary` vs `remeda`?

**Review by bangarang — APPROVED** [review]
Just need some love in the PR description!

**Comment by bangarang** [comment]
Awesome description, and the screenshots are 👩‍🍳 !!

**Comment by bangarang** [comment]
Now that I'm seeing it, I'm debating here is if we should clarify the commands a bit - should they be `delete-agent` (with a some kind of shorthand like `da`), `list-agent `(la)` ,` etc
Wondering if as we add things here, if `list`, `delete` are too vague. We had a customer trying to add namespaces to environments and I'm wondering if some time we'll want to list environments, namespaces, spaces etc...

Thoughts @meritmalling?

**Comment by meritmalling** [comment]
> Now that I'm seeing it, I'm debating here is if we should clarify the commands a bit - should they be `delete -agent` (with a some kind of shorthand like `da`), `list -agents `(`la`) , etc Wondering if as we add things here, if `list` `delete` are too vague. We had a customer trying to add namespaces to environments and I'm wondering if some time we'll want to list environments, namespaces, spaces etc...
> 
> Thoughts @meritmalling?

Really good thinking, we don't want to box ourselves in! I'll update those two!

**Comment by bangarang** [comment]
Closer! But thinking we have a generic `list`/`delete` and then a flag to specify `-agents`(is it better practice here to prefix `-` or `--`) vs other resources. 
For example, Maybe we expand each command to list **all resources** available to the CLI in some kind of tree it out to navigate cleanly.

**Comment by bangarang** [comment]
I think we can revert those last changes and add more list / delete functionality later. Let's get this out the door and add more soon!

**Pull request merged**